backend/ethereum/channel/funder.go

```diff
@@ -336,20 +336,29 @@ func (f *Funder) waitForFundingConfirmation(ctx context.Context, request channel
 			log.Debugf("peer[%d]: got: %v, remaining for [%d,%d] = %v. N: %d", request.Idx, event.Amount, asset.assetIndex, idx, amount, N)
 
 		case <-ctx.Done():
-			var indices []channel.Index
-			for k, bals := range agreement {
-				if bals.Sign() == 1 {
-					indices = append(indices, channel.Index(k))
-				}
-			}
-			if len(indices) != 0 {
-				return &channel.AssetFundingError{Asset: asset.assetIndex, TimedOutPeers: indices}
-			}
-			return nil
+			return fundingTimeoutError(agreement, asset)
 		case err := <-subErr:
-			return err
+			select {
+			case <-ctx.Done():
```

Comment:
As discussed elsewhere, the `select` is more foolproof than the error check, since contexts can fail with two different errors. We decided against the if, but added a comment for explanation.